zict/sieve.py

```diff
@@ -0,0 +1,88 @@
+from __future__ import absolute_import, division, print_function
+
+from collections import MutableMapping
+from itertools import chain
+import sys
+
+
+class Sieve(MutableMapping):
+    """ Store values in different mappings based on their sizes.
+
+    This creates a MutableMapping by combining two MutableMappings,
+    one that stores values strictly smaller (in bytes) than a given
+    threshold, the other storing the other values.
+
+    Only the size of values, not keys, is considered, since keys are
+    likely kept in memory for bookkeeping purposes anyway.
+
+    Parameters
+    ----------
+    small: MutableMapping
+    large: MutableMapping
+    threshold: int
+        The number of bytes below which values are stored in the *small*
+        mapping.
+
+    Examples
+    --------
+    >>> small = {}
+    >>> large = Func(dumps, loads, File('storage/'))  # doctest: +SKIP
+    >>> store = Sieve(small, large, threshold=1024 ** 2)  # doctest: +SKIP
+
+    See Also
+    --------
+    Buffer
+    """
+    def __init__(self, small, large, threshold):
```

Comment:
If there is an arbitrary number of underlying mappings, I think `__setitem__` and `__delitem__` will get considerably more complicated (and potentially slower), since you have to check all mappings for the requested key.